### PR TITLE
Publish prof container master

### PR DESCRIPTION
**Pull request by aksm-ms**

### CODE REVIEW

File: src/ActionInputValidator/ActionValidators/PublishProfileContainerWebAppValidator.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { packageNotAllowed, multiContainerNotAllowed, startupCommandNotAllowed, validateSingleContainerInputs } from "../Validations";
+
```
Comment (asranja) [inline src/ActionInputValidator/ActionValidators/PublishProfileContainerWebAppValidator.ts:2]: extra line

File: src/ActionInputValidator/Validations.ts [hunk 2]
```diff
@@ -63,6 +58,14 @@ export function multiContainerNotAllowed(configFile: string) {
     }
 }
 
+// Error if image name is not provided
+export function validateSingleContainerInputs() {
+    let actionParams: ActionParameters = ActionParameters.getActionParams();
+    if(!actionParams.images) {
+        throw new Error("Image name not provided for a single-container. Provide a valid image name");
```
Comment (asranja) [inline src/ActionInputValidator/Validations.ts:65]: Check all the error messages with PM once.

File: src/DeploymentProvider/Providers/PublishProfileWebAppContainerDeploymentProvider.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { BaseWebAppDeploymentProvider } from './BaseWebAppDeploymentProvider';
+
+export class PublishProfileWebAppContainerDeploymentProvider extends BaseWebAppDeploymentProvider {
+    
+    public async DeployWebAppStep() {
+        let appName: string = this.actionParams.appName;
+        let images: string = this.actionParams.images;
+        // TODO - how to check if its a linux container
```
Comment (asranja) [inline src/DeploymentProvider/Providers/PublishProfileWebAppContainerDeploymentProvider.ts:8]: Get the web app details and check if there is any property which indicates if it is a linux type.

File: src/DeploymentProvider/DeploymentProviderFactory.ts [hunk 4]
```diff
@@ -4,14 +4,20 @@ import { DEPLOYMENT_PROVIDER_TYPES } from "./Providers/BaseWebAppDeploymentProvi
 import { IWebAppDeploymentProvider } from "./Providers/IWebAppDeploymentProvider";
 import { WebAppContainerDeploymentProvider } from "./Providers/WebAppContainerDeployment";
 import { WebAppDeploymentProvider } from "./Providers/WebAppDeploymentProvider";
+import { PublishProfileWebAppContainerDeploymentProvider } from "./Providers/PublishProfileWebAppContainerDeploymentProvider";
 
 export class DeploymentProviderFactory {
 
     public static getDeploymentProvider(type: DEPLOYMENT_PROVIDER_TYPES) : IWebAppDeploymentProvider {
         
         // For publish profile type app kind is not available so we directly return WebAppDeploymentProvider
```
Comment (asranja) [inline src/DeploymentProvider/DeploymentProviderFactory.ts:13]: edit this comment.

File: src/DeploymentProvider/DeploymentProviderFactory.ts [hunk 5]
```diff
@@ -4,14 +4,20 @@ import { DEPLOYMENT_PROVIDER_TYPES } from "./Providers/BaseWebAppDeploymentProvi
...
 
     public static getDeploymentProvider(type: DEPLOYMENT_PROVIDER_TYPES) : IWebAppDeploymentProvider {
         
         // For publish profile type app kind is not available so we directly return WebAppDeploymentProvider
         if(type === DEPLOYMENT_PROVIDER_TYPES.PUBLISHPROFILE) {
-            return new WebAppDeploymentProvider(type);
+            if (!!ActionParameters.getActionParams().images) {
+                return new PublishProfileWebAppContainerDeploymentProvider(type);
+            }
+            else{
```
Comment (asranja) [inline src/DeploymentProvider/DeploymentProviderFactory.ts:18]: space after else

File: src/ActionInputValidator/ActionValidators/PublishProfileContainerWebAppValidator.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    async validate(): Promise<void> {
+
+        let actionParams: ActionParameters = ActionParameters.getActionParams();
+
+        packageNotAllowed(actionParams.packageInput);
+
+        // TODO: windows not allowed 
+
+        multiContainerNotAllowed(actionParams.multiContainerConfigFile);
+
```
Comment (asranja) [inline src/ActionInputValidator/ActionValidators/PublishProfileContainerWebAppValidator.ts:17]: extra line

File: src/ActionInputValidator/ActionValidators/PublishProfileContainerWebAppValidator.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+export class PublishProfileContainerWebAppValidator implements IValidator {
+    
+    async validate(): Promise<void> {
+
+        let actionParams: ActionParameters = ActionParameters.getActionParams();
+
+        packageNotAllowed(actionParams.packageInput);
+
+        // TODO: windows not allowed
```
Comment (asranja) [inline src/ActionInputValidator/ActionValidators/PublishProfileContainerWebAppValidator.ts:14]: TODO

File: src/ActionInputValidator/ActionValidators/PublishProfileContainerWebAppValidator.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import { packageNotAllowed, multiContainerNotAllowed, startupCommandNotAllowed, validateSingleContainerInputs } from "../Validations";
+
+import { ActionParameters } from "../../actionparameters";
+import { IValidator } from "./IValidator";
+
+export class PublishProfileContainerWebAppValidator implements IValidator {
+    
+    async validate(): Promise<void> {
```
Comment (asranja) [inline src/ActionInputValidator/ActionValidators/PublishProfileContainerWebAppValidator.ts:8]: public async

File: src/DeploymentProvider/Providers/PublishProfileWebAppContainerDeploymentProvider.ts (hunk 3 above)
Comment (aksm-ms) [reply src/DeploymentProvider/Providers/PublishProfileWebAppContainerDeploymentProvider.ts:8]: Working on getting web app details using publish profile details.

File: src/ActionInputValidator/ActionValidators/PublishProfileContainerWebAppValidator.ts (hunk 8 above)
Comment (aksm-ms) [reply src/ActionInputValidator/ActionValidators/PublishProfileContainerWebAppValidator.ts:8]: public by default - https://www.typescriptlang.org/docs/handbook/classes.html#public-by-default

Comment (asranja) [reply src/ActionInputValidator/ActionValidators/PublishProfileContainerWebAppValidator.ts:6]: It is just that we have been using public async in all actions. Wanted to keep it in sync. :)

**Review by asranja — APPROVED** [review]

**Pull request merged**